sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -63,6 +63,61 @@ class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
     }
   }
 
+  override def tableExists(ident: Identifier): Boolean = {
+    if (ident.namespace().length <= 1) {
+      catalog.tableExists(ident.asTableIdentifier)
+    } else {
+      false
+    }
+  }
+
+  private def convertV1TableToV2(provider: TableProvider, v1Table: CatalogTable): Table = {
+    val options: Map[String, String] = {
+      v1Table.storage.locationUri match {
+        case Some(uri) =>
+          v1Table.storage.properties + ("path" -> uri.toString)
+        case _ =>
+          v1Table.storage.properties
+      }
+    }
+
+    val partitioning: Array[Transform] = {
+      val partitions = new mutable.ArrayBuffer[Transform]()
+
+      v1Table.partitionColumnNames.foreach { col =>
+        partitions += LogicalExpressions.identity(col)
```

Comment:
Nit: This parses the column's name as a multi-part identifier, which is subtly incorrect. (It'll cause issues if the column name contains special characters like ':'.)

Reply:
Aren't column names with special characters supported? I thought you could escape any identifier using back-ticks.